rdkcentral/Lightning-UI-Components
# fix(NavigationManager): allow appending items without overwriting stored lazy items

## Description
- abstracts `appendItems`, `appendItemsAt`, `lazyUpCount`, and `lazyUpCountBuffer`  from `Row` and `Column` to `NavigationManager` for shared functionality
- if an item is appended after the index of the last visible item, it will be added to lazy items and not immediately displayed

## Testing
Do the following in both of these Row and Column stories:
- [http://localhost:8000/?path=/story/navigation-row--adding-items](http://localhost:8000/?path=/story/navigation-row--adding-items)
- [http://localhost:8000/?path=/story/navigation-column--adding-items](http://localhost:8000/?path=/story/navigation-column--adding-items)
1. set `lazyUpCount` to a number less than the number of total items that exist in the Row/Column (less than 12)
2. refresh the story so lazy upcount functionality exists
3. after 3 seconds 3 buttons will be appended at index 3.
ER: If buttons at index 3 were not visible due to the `lazyUpCount`, they should remain not visible until navigating to those buttons
**NOTE: This does not currently apply to prepending items, as `lazyItems` are only added to the visible items when navigating "forward" through the Row/Column (`selectNextMethod`). Prepended items will be displayed immediately.**
